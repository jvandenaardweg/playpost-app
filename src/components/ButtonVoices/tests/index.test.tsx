import React from 'react';
import { fireEvent, render, RenderAPI } from 'react-native-testing-library';

import { ButtonVoices } from '../index';

describe('ButtonVoices', () => {
  let wrapper: RenderAPI;
  const onPressHandler = jest.fn();

  describe('rendering', () => {
    beforeAll(() => {
      wrapper = render(<ButtonVoices onPress={onPressHandler} />);
    });

    it('should render correctly', () => {
      expect(wrapper.toJSON()).toMatchSnapshot();
    });
  });

  describe('interacting', () => {
    beforeAll(() => {
      wrapper = render(<ButtonVoices onPress={onPressHandler} />);
    });

    it('should fire onPress when the button is pressed', () => {
      fireEvent.press(wrapper.getByTestId('ButtonVoices-button'));
      expect(onPressHandler).toHaveBeenCalledTimes(1);
    });
  });
});
