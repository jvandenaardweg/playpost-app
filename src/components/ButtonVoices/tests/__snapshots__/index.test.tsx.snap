// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonVoices rendering should render correctly 1`] = `
<View
  style={
    Object {
      "paddingLeft": 12,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 28,
          "height": 28,
        },
        Object {
          "backgroundColor": "#e5e5e5",
        },
      ]
    }
    testID="ButtonTiny-Button"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 28,
            "flexDirection": "row",
            "height": 28,
            "paddingLeft": 12,
            "paddingRight": 12,
          },
          Object {
            "backgroundColor": "#e5e5e5",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#090909",
              "fontFamily": "Inter-SemiBold",
              "fontSize": 13,
              "lineHeight": 16,
            },
            Object {
              "color": "#090909",
            },
          ]
        }
      >
        Voices
      </Text>
    </View>
  </View>
</View>
`;
